### ISSUE TITLE
App Header, App Footer, Side Menu, One Column Layout: Use proper HTML Landmarks

### ISSUE THREAD

**Issue by ArakTaiRoth** [GovAlta/ui-components#2362]
# Info

This was raised in the UAIM Accessibility Reported as a Level 3 Issue. The issue is as reported, "Regions (footer, etc.) not labelled correctly; could easily get lost and not really have a mental map of the site". The recommendation is to use proper descriptive landmarks to help users understand the structure and navigate more effectively.

## Acceptance Criteria:

1. Update effected components to use proper landmarks within the component
    * App Header --> `<header>` and `<nav>` landmarks
    * App Footer --> '<footer>` landmark
    * Side Menu --> `<nav>` landmark
    * One Column Layout --> `<main>` landmark

**Comment by Spark450** [comment]
Do we need to consider [unique nav labels for situations where we have multiple nav elements on the page at one time](https://www.w3.org/WAI/ARIA/apg/patterns/landmarks/examples/navigation.html#:~:text=HTML%20Example%3A%20More%20Than%20One%20Navigation%20Landmark%20Example).
